physics/noahmpdrv.F90

```diff
@@ -88,6 +89,8 @@ subroutine noahmpdrv_init(lsm, lsm_noahmp, me, isot, ivegsrc, &
         !--- initialize soil vegetation
         call set_soilveg(me, isot, ivegsrc, nlunit, errmsg, errflg)
 
+        !--- read in noahmp table
+        call read_mp_table_parameters
```

Comment:
I suggest passing in control information to read_mp_table_parameters() and improving the error handling. 
- You only need to read in the data on the master processor.
- All of the namelist messages should be output if there is a failure.